util/xz.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"io"
+	"os"
+
+	"github.com/ulikunitz/xz"
+)
+
+// XZ2File does xz decompression from src file into dst file
+func XZ2File(dst, src string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	defer in.Close()
+
+	out, err := os.OpenFile(dst, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+
+	reader, err := xz.NewReader(in)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(out, reader)
+	if err != nil {
+		os.Remove(dst)
+	}
+	return err
```

Comment:
In general, avoid doing this, as `error` is an interface: https://golang.org/doc/faq#nil_error

Just return the error in the conditional above and `return nil` here.